cpp/src/decisiontree/decisiontree_impl.cuh

```diff
@@ -214,25 +214,24 @@ template <typename T, typename L>
 void DecisionTreeBase<T, L>::plant(
   std::vector<SparseTreeNode<T, L>> &sparsetree, const T *data, const int ncols,
   const int nrows, const L *labels, unsigned int *rowids,
-  const int n_sampled_rows, int unique_labels, const int treeid, int maxdepth,
-  int max_leaf_nodes, const float colper, int n_bins, int split_algo_flag,
-  int cfg_min_rows_per_node, bool cfg_bootstrap_features,
-  CRITERION cfg_split_criterion, bool quantile_per_tree) {
-  split_algo = split_algo_flag;
+  const int n_sampled_rows, int unique_labels, const int treeid,
+  DecisionTreeParams &tree_params) {
+  split_algo = tree_params.split_algo;
   dinfo.NLocalrows = nrows;
   dinfo.NGlobalrows = nrows;
   dinfo.Ncols = ncols;
-  nbins = n_bins;
-  treedepth = maxdepth;
-  maxleaves = max_leaf_nodes;
+  nbins = tree_params.n_bins;
+  treedepth = tree_params.max_depth;
+  maxleaves = tree_params.max_leaves;
   n_unique_labels = unique_labels;
-  min_rows_per_node = cfg_min_rows_per_node;
-  bootstrap_features = cfg_bootstrap_features;
-  split_criterion = cfg_split_criterion;
+  min_rows_per_node = tree_params.min_rows_per_node;
+  bootstrap_features = tree_params.bootstrap_features;
+  split_criterion = tree_params.split_criterion;
```

Comment:
minor nitpick: do we need these assignments? Why not directly access these from the `tree_params` itself?

Reply:
These are assigned to members of the object. I am not sure if they are used elsewhere or not.

Reply:
ah.. ok. Then what do you think of instead creating a member variable of type `DecisionTreeParams` itself? (basically extending your idea of removing extra params to even the class members)

Reply:
That sounds good, let me quickly see how that would look like.